01-data-structures/05-hashes-part-2/separate_chaining/separate_chaining.rb

```diff
@@ -1,33 +1,85 @@
+require_relative 'node'
 require_relative 'linked_list'
 
 class SeparateChaining
   attr_reader :max_load_factor
 
   def initialize(size)
     @max_load_factor = 0.7
+    @items = Array.new(size)
   end
 
   def []=(key, value)
+    node = Node.new(key, value)
+    i = index(key, self.size)
+    if @items[i].nil?
+      item = LinkedList.new
+      item.add_to_tail(node)
+      @items[i] = item
+    elsif @items[i].size < 5
+      @items[i].add_to_tail(node)
+    else
+      resize
+      self[key] = value
+    end
```

Comment:
The `elsif` condition if handy for efficiency, but the `load_factor` should actually be handling this on average. You could just make in an `else`. With that, this method can actually be condensed to something like this:

```ruby
i = self.index(key, self.size)
list = @items[i]
if list.nil?
  list = LinkedList.new
  @items[i] = list
end
list.add_to_tail(Node.new(key, value)
resize if self.load_factor > @max_load_factor
```